x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Replace.java

```diff
@@ -121,15 +125,15 @@ public boolean foldable() {
         return str.foldable() && regex.foldable() && newStr.foldable();
     }
 
-    @Evaluator(extraName = "Constant", warnExceptions = PatternSyntaxException.class)
+    @Evaluator(extraName = "Constant", warnExceptions = IllegalArgumentException.class)
```

Comment:
`PatternSyntaxException` is an `IllegalArgumentException`. If we choose another exception for this PR, we would have to restore `PatternSyntaxException`